tools/tag-release

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import shutil
+from subprocess import run
+import sys
+import textwrap
+
+
+def parse_tag(tag_name):
+    """Parse the version from its tag name.
+
+    Release tags are of the form "v0.6.3", and release candidates are of the form "v0.6.3-rc2".
+
+    Raise a ValueError if the tag isn't in a recognised format.
+    """
+    pattern = re.compile(
+        r"""
+            v
+            (?P<major>\d+)
+            .
```

Comment:
minor point but I think these `.` should be `\.` so they only match full stops rather than any character

Reply:
gotta love regexps